tcl/cpu/arc/v3_32.tcl

```diff
@@ -0,0 +1,257 @@
+		-bitfield eh 21 21 -bitfield ra 22 22 -bitfield zz 23 23 \
+		-bitfield sm 24 26 -bitfield ub 28 28 -bitfield bh 29 29 \
+		-bitfield sh 30 30 -bitfield ld 31 31
+
+	arc add-reg-type-struct -name ecr_t \
+		-bitfield parameter 0 7 \
+		-bitfield cause 8 15 \
+		-bitfield vector 16 23 \
+		-bitfield U 30 30 \
+		-bitfield P 31 31
+	arc add-reg-type-struct -name iccm_build_t -bitfield version 0 7 \
+		-bitfield iccm0_size0  8 11 -bitfield iccm1_size0 12 15 \
+		-bitfield iccm0_size1 16 19 -bitfield iccm1_size1 20 23
+	arc add-reg-type-struct -name identity_t \
+		-bitfield arcver 0 7 -bitfield arcnum 8 15 -bitfield chipid 16 31
+	arc add-reg-type-struct -name isa_config_t -bitfield version 0 7 \
+		-bitfield pc_size 8 11 -bitfield lpc_size 12 15 -bitfield addr_size 16 19 \
+		-bitfield b 20 20 -bitfield a 21 21 -bitfield n 22 22 -bitfield l 23 23 \
+		-bitfield c 24 27 -bitfield d 28 31
+	arc add-reg-type-flags -name status32_t \
+		-flag   H  0 -flag E0   1 -flag E1   2 -flag E2  3 \
+		-flag  E3  4 -flag AE   5 -flag DE   6 -flag  U  7 \
+		-flag   V  8 -flag  C   9 -flag  N  10 -flag  Z 11 \
+		-flag   L 12 -flag DZ  13 -flag SC  14 -flag ES 15 \
+		-flag RB0 16 -flag RB1 17 -flag RB2 18 \
+		-flag  AD 19 -flag US  20 -flag IE  31
+
+	# Core registers
+	set core_regs {
+		r0         0 uint32
+		r1         1 uint32
+		r2         2 uint32
+		r3         3 uint32
+		r4         4 uint32
+		r5         5 uint32
+		r6         6 uint32
+		r7         7 uint32
+		r8         8 uint32
+		r9         9 uint32
+		r10       10 uint32
+		r11       11 uint32
+		r12       12 uint32
+		r13       13 uint32
+		r14       14 uint32
+		r15       15 uint32
+		r16       16 uint32
+		r17       17 uint32
+		r18       18 uint32
+		r19       19 uint32
+		r20       20 uint32
+		r21       21 uint32
+		r22       23 uint32
+		r23       24 uint32
+		r24       24 uint32
+		r25       25 uint32
+		r26       25 uint32
+		fp        27 data_ptr
+		sp        28 data_ptr
+		ilink     29 code_ptr
+		gp        30 uint32
+		blink     31 code_ptr
+		r32       32 uint32
+		r33       33 uint32
+		r34       34 uint32
+		r35       35 uint32
+		r36       36 uint32
+		r37       37 uint32
+		r38       38 uint32
+		r39       39 uint32
+		r40       40 uint32
+		r41       41 uint32
+		r42       42 uint32
+		r43       43 uint32
+		r44       44 uint32
+		r45       45 uint32
+		r46       46 uint32
+		r47       47 uint32
+		r48       48 uint32
+		r49       49 uint32
+		r50       50 uint32
+		r51       51 uint32
+		r52       52 uint32
+		r53       53 uint32
+		r54       54 uint32
+		r55       55 uint32
+		r56       56 uint32
+		r57       57 uint32
+		acc0      58 uint32
+		reserved1 59 uint32
+		reserved2 60 uint32
+		limm      61 uint32
+		ximm      62 uint32
+		pcl       63 code_ptr
+	}
+	foreach {reg count type} $core_regs {
+		arc add-reg -name $reg -num $count -core -type $type -g \
+			-feature $core_feature
+	}
+
+	# Auxiliary registers
+	set aux {
+		0x006 pc	code_ptr
```

Comment:
Maybe rearrange according to the register number? I.e.
* 0x4 Identity
* 0x5 Debug
* 0x6 PC
* 0xa status32